app/models/teleconsultation.rb

```diff
@@ -0,0 +1,42 @@
+class Teleconsultation < ApplicationRecord
+  include Mergeable
+
+  belongs_to :patient
+  belongs_to :medical_officer, class_name: "User", foreign_key: :medical_officer_id
```

Comment:
Does this need to be optional? We won't have it when the request comes in first right?

Reply:
We have this information in the request also since the nurse knows which MO the teleconsult is sent to.

Reply:
I seeeeeee, so the MO ID sent in the request payload is "who has been asked to do the teleconsult".

Let me address [my previous comment](https://github.com/simpledotorg/simple-server/pull/1316/files#r490089933) here as well in one big bundle then.

Conceptually, there's a difference between the MO that's _requested_ to do the teleconsult, and the MO that _does_ the teleconsult.

If we want them to be the same, we should enforce that when we receive a `record` from the MO. This is a stricter model, not allowing, for example, MOs to forward these links to each other ("I'm busy, can you take this one").

In the more lenient model, the MO requested doesn't have to be the same as the MO who does the teleconsult. However, in that case, these are two separate concepts that we should capture distinctly. We can rename this to `requested_medical_officer` or something to make it very clear, and introduce a `medical_officer` association that stores which MO actually did the teleconsultation. Our API can do this in a fairly straightforward manner:

```ruby
unless teleconsult.completed?
  payload.merge(medical_officer_id: request_user.id)
end
```

This wont' require any API changes I believe. This will also allow us to make smart inferences about the usage of telemedicine - eg. how much link sharing is going on, _can_ we enforce the stricter model in the future, etc.

What do you think? Let me know if you wanna chat.

Reply:
Ah, that makes total sense. Yes, imo we should introduce a check here, instead of allowing `requested_medical_officer` and `medical_officer` to be separate things. Something like:

```
authorise record update if payload["medical_officer_id"] = request_user.id
```

Nurses can send requests for the same patient to many doctors in the first place, so I think we shouldn't support/instrument for the lenient model.

Reply:
Added the `requested_medical_officer` in 0beb9d5

Reply:
To close the loop, we decided to go with the lenient `requested_medical_officer` approach so we can first learn how big or small this problem is, before introducing constraints into the system. Additionally, constraining MOs on the serverside cannot be mirrored by a constraint in the app today, so it would lead to a misleading user experience where a seemingly successful teleconsultation would be rejected by the server.